compose.js

```diff
@@ -88,12 +92,10 @@ function mergeComposable(dstDescriptor, srcComposable) {
   combineProperty('configuration', assign);
   combineProperty('deepConfiguration', merge);
   if (Array.isArray(srcDescriptor.initializers)) {
-    dstDescriptor.initializers = srcDescriptor.initializers.reduce((result, init) => {
-      if (isFunction(init) && result.indexOf(init) < 0) {
-        result.push(init);
-      }
-      return result;
-    }, Array.isArray(dstDescriptor.initializers) ? dstDescriptor.initializers : []);
+    const initializers = (dstDescriptor.initializers || [])
```

Comment:
No it does not answer it because moving this to checking to `createFactory` isn't certainly going to break any tests as you foretold. Sorry, but I don't see any added value in duplicating these checks.